src/service/artifacts/artifact-service.ts

```diff
@@ -0,0 +1,59 @@
+import Container, { Service } from "typedi";
+import { Node } from "@bc/domain/node";
+import { ConfigurationService } from "@bc/service/config/configuration-service";
+import { UploadService } from "@bc/service/artifacts/upload-service";
+import { ArchiveDependencies } from "@bc/domain/archive";
+import { LoggerService } from "@bc/service/logger/logger-service";
+import { LoggerServiceFactory } from "@bc/service/logger/logger-service-factory";
+import { UploadResponse } from "@actions/artifact";
+
+@Service()
+export class ArtifactService {
+  private readonly configService: ConfigurationService;
+  private readonly uploadService: UploadService;
+  private readonly logger: LoggerService;
+
+  constructor() {
+    this.configService = Container.get(ConfigurationService);
+    this.uploadService = Container.get(UploadService);
+    this.logger = LoggerServiceFactory.getInstance();
+  }
+
+  private getNodesToArchive(nodeChain: Node[]): Node[] {
```

Comment:
it is my recommendation to have one single exit/return point for almost every method, specially when have more than two or you can change it by a switch one, where like exit points are a bit more clear. And additionally why don't you use `this.configService.isNodeStarter`?. So I would suggest
```
  private getNodesToArchive(nodeChain: Node[]): Node[] {
    const startingProject = this.configService.getStarterNode();
    const dependencies = startingProject.archiveArtifacts?.dependencies ?? ArchiveDependencies.NONE;
    let result = undefined;
    if (dependencies === ArchiveDependencies.NONE) {
      result = [this.configService.getStarterNode()];
    } else if (dependencies === ArchiveDependencies.ALL) {
      result = nodeChain.filter((node) => !!node.archiveArtifacts);
    } else {
      result = nodeChain.filter((node) => node.archiveArtifacts && (dependencies.includes(node.project) || this.configService.isNodeStarter(node));
    }
    return result;
  }
```
or
```
  private getNodesToArchive(nodeChain: Node[]): Node[] {
    const dependencies = startingProject.archiveArtifacts?.dependencies ?? ArchiveDependencies.NONE;
    switch(dependencies) {
      case ArchiveDependencies.NONE:
       return [this.configService.getStarterNode()];
      case ArchiveDependencies.ALL:
       return nodeChain.filter((node) => !!node.archiveArtifacts);
      default:
       return nodeChain.filter((node) => node.archiveArtifacts && (dependencies.includes(node.project) || this.configService.isNodeStarter(node));
    }
  }
```